server/src/main/java/io/druid/initialization/Initialization.java

```diff
@@ -143,25 +122,39 @@ static void clearLoadedModules()
    */
   public synchronized static <T> Collection<T> getFromExtensions(ExtensionsConfig config, Class<T> clazz)
   {
-    final TeslaAether aether = getAetherClient(config);
     final Set<T> retVal = Sets.newHashSet();
-    final Set<String> extensionNames = Sets.newHashSet();
+    final Set<String> loadedExtensionNames = Sets.newHashSet();
 
-    for (String coordinate : config.getCoordinates()) {
-      log.info("Loading extension[%s] for class[%s]", coordinate, clazz.getName());
-      try {
-        URLClassLoader loader = getClassLoaderForCoordinates(aether, coordinate, config.getDefaultVersion());
+    if (config.searchCurrentClassloader()) {
+      for (T module : ServiceLoader.load(clazz, Thread.currentThread().getContextClassLoader())) {
```

Comment:
ServiceLoader.load checks context classloader automatically, but it probably doesn't hurt to have it here.